sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py

```diff
@@ -349,25 +359,39 @@ def send_messages(self, message, **kwargs):
         timeout = kwargs.pop("timeout", None)
         if timeout is not None and timeout <= 0:
             raise ValueError("The timeout must be greater than 0.")
-        message = transform_messages_to_sendable_if_needed(message)
-        try:
-            batch = self.create_batch()
-            batch._from_list(message)  # pylint: disable=protected-access
-            message = batch
-        except TypeError:  # Message was not a list or generator.
-            pass
-        if isinstance(message, BatchMessage) and len(message) == 0:  # pylint: disable=len-as-condition
-            raise ValueError("A BatchMessage or list of Message must have at least one Message")
-        if not isinstance(message, BatchMessage) and not isinstance(message, Message):
-            raise TypeError("Can only send azure.servicebus.<BatchMessage,Message> or lists of Messages.")
-
-        self._do_retryable_operation(
-            self._send,
-            message=message,
-            timeout=timeout,
-            operation_requires_timeout=True,
-            require_last_exception=True
-        )
+
+        with send_trace_context_manager() as send_span:
+            # Ensure message is sendable (not a ReceivedMessage), and if needed (a list) is batched. Adds tracing.
+            message = transform_messages_to_sendable_if_needed(message)
+            try:
+                for each_message in iter(message): # type: ignore # Ignore type (and below) as it will except if wrong.
+                    add_link_to_send(each_message, send_span)
```

Comment:
is `trace_message(message, send_span)` needed here? as I see in `_build_schedule_request` these two methods call come in pair